tests/python_api/external_source_reader.py

```diff
@@ -7,14 +7,17 @@
 import numpy as np
 import cupy as cp
```

Comment:
move this to line 44 if statement. We do not want cpu to fail if cupy is not available
```
if device == "gpu":
```

Reply:
@swetha097 can you please resolve this? i cannot run this sample without cupy on my system otherwise

Reply:
Done